### ISSUE TITLE
Built-in functions

### ISSUE THREAD

**Issue by QinshiWang** [verified-network-toolchain/petr4#302]
Currently, the semantics of built-in functions only supports the case that the "this" object to be an lvalue. Maybe we want to support rvalue in some cases. If so, then I think we should convert built-in function calls into kind of normal function calls with built-in operations. For example, we can have function prototypes
```
bool isValid(in hdr);
void setValid(inout hdr);
void setInvalid(inout hdr);
```
And the calls are converted as
```
hdr.isValid() -> isValid(hdr)
hdr.setValid() -> setValid(hdr)
hdr.setInvalid() -> setInvalid(hdr)
```

**Comment by hackedy** [comment]
Yeah I think being able to place a direction annotation on the `this` parameter would be useful.
